sdk/storage/azure-storage-common/pom.xml

```diff
@@ -58,12 +58,6 @@
       <scope>provided</scope>
     </dependency>
 
-    <dependency>
```

Comment:
Don't we need the azure-core `test-jar` dependency here now that `azure-core-test` is removed? Or was this dependency added eagerly before it was required?